atomic_reactor/plugins/build_orchestrate_build.py

```diff
@@ -650,13 +674,167 @@ def set_build_image(self):
             raise RuntimeError("Build object is malformed, failed to fetch buildroot image")
 
         if build_kind == 'DockerImage':
-            self.config_kwargs['build_image'] = build_name
+            return build_name
         else:
             raise RuntimeError("Build kind isn't 'DockerImage' but %s" % build_kind)
 
+    def process_image_from(self, image_from):
+        build_image = None
+        imagestream = None
+
+        if image_from['kind'] == 'DockerImage':
+            build_image = image_from['name']
+        elif image_from['kind'] == 'ImageStreamTag':
+            imagestream = image_from['name']
+        else:
+            raise UnknownKindException
+
+        return build_image, imagestream
+
+    def set_build_image(self, platforms):
+        """
+        Overrides build_image for worker, to be same as in orchestrator build
+        """
+        current_platform = platform.processor()
+        orchestrator_platform = current_platform if current_platform else 'x86_64'
+        status = get_build_json().get("status")
+        build_config_name = None
+        build_image = None
+        imagestream = None
+        need_manifest_list = not (orchestrator_platform in platforms and len(platforms) == 1)
+        use_same_image = False
+        current_buildimage = self.get_current_buildimage()
+
+        # BuildConfig exists
+        if 'config' in status:
+            config = status['config']
+            if config['kind'] == 'BuildConfig':
+                build_config_name = config['name']
+            else:
+                raise RuntimeError("Build config type isn't BuildConfig : %s" % config['kind'])
+
+        # get image build from build metadata, which is set for direct builds
```

Comment:
Let's clarify that this is "set by osbs-client" to avoid confusion with OpenShift behavior.

Reply:
ok